javasource/restservices/util/JsonDeserializer.java

```diff
@@ -66,101 +66,180 @@ else if (jsonValue instanceof Long || jsonValue instanceof Double || jsonValue i
+						IMendixIdentifier child = readJsonDataIntoMendixObject(context, children.get(i), otherSideType, autoResolve);
+						if (child != null) {
+							/*
+							 * The core.createMendixIdentifier should be unnecessary, however, there is a bug there, see
+							 * support ticket 102188
+							 */
+							ids.add(Core.createMendixIdentifier(child.toLong()));
+						}
 					}
+
+					((MendixObjectReferenceSet)member).setValue(context, ids);
 				}
-				
-				((MendixObjectReferenceSet)member).setValue(context, ids);
+
+				//Primitive member
+				else
+				{
+					IMetaPrimitive primitive = target.getMetaObject().getMetaPrimitive(targetattr);
+					if (primitive.getType() != PrimitiveType.AutoNumber)
+						target.setValue(context, targetattr, jsonAttributeToPrimitive(primitive, object, attr));
+				}
+
 			}
-			
-			//Primitive member
-			else if (target.hasMember(targetattr)){
-				IMetaPrimitive primitive = target.getMetaObject().getMetaPrimitive(targetattr);
-				if (primitive.getType() != PrimitiveType.AutoNumber)
-					target.setValue(context, targetattr, jsonAttributeToPrimitive(primitive, object, attr));
+			else
+			{
+				IMetaAssociation association =  Core.getMetaAssociation(targetattr);
+				String otherSideType = association.getParent().getName();
+				IMendixObject otherSideTarget = Core.instantiate(context, otherSideType);
+				Map<String, String> otherAttributeNameMap = buildAttributeNameMap(otherSideTarget.getMetaObject());
+				String otherTargetattr =  otherAttributeNameMap.get(attr.toLowerCase().replaceAll("[^a-zA-Z0-9_]","_"));
+
+				IMendixObjectMember<?> otherMember = otherSideTarget.getMember(context, otherTargetattr);
+
+				//Reference for arrays
+				if (otherMember instanceof MendixObjectReference)
+				{
+					JSONArray jsonArray = object.getJSONArray(attr);
+					readJsonArrayIntoMendixObject(context, jsonArray, otherSideTarget, autoResolve, association.getName().split("\\.")[1], target.getId());
+				}
 			}
 		}
 		Core.commit(context, target);
 	}
 
+	private static void readJsonArrayIntoMendixObject(IContext context, JSONArray object, IMendixObject target, boolean autoResolve, String associationName, IMendixIdentifier mainIdentifier) throws JSONException, Exception {
+
+		Map<String, String> attributeNameMap = buildAttributeNameMap(target.getMetaObject());
+
+		String targetattr =  attributeNameMap.get(associationName.toLowerCase().replaceAll("[^a-zA-Z0-9_]","_"));
+		IMetaAssociation association =  Core.getMetaAssociation(targetattr);
+		String otherSideType = association.getParent().getName();
+
+		for(int i = 0; i < object.length(); i++)
+		{
+			IMendixObject otherSideTarget = Core.instantiate(context, otherSideType);
+			IMendixObjectMember<?> otherMember = otherSideTarget.getMember(context, targetattr);
+			((MendixObjectReference)otherMember).setValue(context, mainIdentifier);
+
+			Object childJsonValue = object.get(i);
+			if(childJsonValue instanceof JSONArray)

```

Comment:
we are dealing with a double nested array here? In that case that deserves some attention in the docs I think about how they are handled